### ISSUE TITLE
'echo test | docker run -i busybox cat' - doesn't succeed

### ISSUE THREAD

**Issue by vburenin** [vmware/vic#3017]
Actual behavior

```
➜  vic git:(master) ✗ echo 'test' | docker run -i busybox
Unable to find image 'busybox:latest' locally
Pulling from library/busybox
56bec22e3559: Pull complete
a3ed95caeb02: Pull complete
Digest: sha256:98a0bd48d22ff96ca23bfda2fe1cf72034ea803bd79e64a5a5f274aca0f9c51c
Status: Downloaded newer image for library/busybox:latest
Error response from daemon: Server error from portlayer: ContainerWaitHandler(187537f9e81de0478c796eb81ecb7e70f0c80d762212854d5150dde5dcfba50a) Error: context deadline exceeded
```

Expected behavior

```
~ » echo test | docker run -i busybox cat
Unable to find image 'busybox:latest' locally
latest: Pulling from library/busybox
56bec22e3559: Pull complete
Digest: sha256:29f5d56d12684887bdfa50dcd29fc31eea4aaf4ad3bec43daf19026a7ce69912
Status: Downloaded newer image for busybox:latest
test
```

**Comment by vburenin** [comment]
[container-logs.zip](https://github.com/vmware/vic/files/572125/container-logs.zip)

**Comment by mlh78750** [comment]
cc @caglar10ur related to #2660

**Comment by caglar10ur** [comment]
dup of #2660

**Comment by caglar10ur** [comment]
re-opening this fixing #2660 won't fix this

**Comment by mdubya66** [comment]
why do we think this is critical for v0.8?

**Comment by caglar10ur** [comment]
This lets users to pipe things into the container, could be helpful for scripting interactions

```echo "caglar" | docker run busybox wc -c``` for example will return 7. Not sure whether it is critical for 0.8.

It heavily depends on #2660 and while testing, I realized fixing that won't fix this (This requires additional synchronization between reading from stdin and launching the process)

My personal suggestion would be marking this "good to have"

**Comment by mlh78750** [comment]
If it doesn't make it to 0.8 we need to put this in known issues. This is something that will likely bite folks that are "kicking the tires" but not much else.

**Comment by mdubya66** [comment]
it's not going to make it. tagging it for release notes.

**Comment by corrieb** [comment]
This shouldn't be low priority IMHO. This is a compatibility issue that should be addressed. Verified that we still don't handle input being piped to a container. Changing to medium.

**Comment by mdubya66** [comment]
This seems to work now. Closing as fixed. 
```
(mwilliamson) ib-aus-office-124:~/Downloads/vic-14308 $ echo test | docker --tls run -i busybox cat
Unable to find image 'busybox:latest' locally
latest: Pulling from library/busybox
03b1be98f3f9: Pull complete
a3ed95caeb02: Pull complete
Digest: sha256:030fcb92e1487b18c974784dcc110a93147c9fc402188370fbfd17efabffc6af
Status: Downloaded newer image for library/busybox:latest
test
(mwilliamson) ib-aus-office-124:~/Downloads/vic-14308 $ (mwilliamson) ib-aus-office-124:~/Downloads/vic-14308 $ echo test | docker --tls run -i busybox cat
(mwilliamson) ib-aus-office-124:~/Downloads/vic-14308 $ docker --tls info
Containers: 1
 Running: 0
 Paused: 0
 Stopped: 1
Images: 1
Server Version: v1.2.0-rc1-14308-8812adb
Storage Driver: vSphere Integrated Containers v1.2.0-rc1-14308-8812adb Backend Engine
VolumeStores: default
vSphere Integrated Containers v1.2.0-rc1-14308-8812adb Backend Engine: RUNNING
 VCH CPU limit: 2942 MHz
 VCH memory limit: 1.242 GiB
 VCH CPU usage: 23 MHz
 VCH memory usage: 827.6 MiB
 VMware Product: VMware ESXi
 VMware OS: vmnix-x86
 VMware OS version: 6.5.0
 Registry Whitelist Mode: disabled.  All registry access allowed.
Plugins:
 Volume: vsphere
 Network: bridge
 Log:
Swarm: inactive
Operating System: vmnix-x86
OSType: vmnix-x86
Architecture: x86_64
CPUs: 2942
Total Memory: 1.242GiB
ID: vSphere Integrated Containers
Docker Root Dir:
Debug Mode (client): false
Debug Mode (server): false
Registry: registry.hub.docker.com
Experimental: false
Live Restore Enabled: false
```
